src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,107 @@
+            statement.setBigDecimal(PRICE_POSITION, book.getPrice());
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                Long id = resultSet.getObject(ID_INDEX, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Connection connection = ConnectionDB.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(SELECT_BY_ID)) {
+            statement.setLong(ID_INDEX, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(createBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find book with id: " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Connection connection = ConnectionDB.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(SELECT_ALL)) {
+            ResultSet resultSet = statement.executeQuery();
+            List<Book> result = new ArrayList<>();
+            while (resultSet.next()) {
+                result.add(createBook(resultSet));
+            }
+            return result;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find all books", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionDB.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(UPDATE)) {
+            statement.setString(TITLE_POSITION, book.getTitle());
+            statement.setObject(PRICE_POSITION, book.getPrice());
+            statement.setLong(ID_POSITION, book.getId());
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionDB.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(DELETE_BY_ID)) {
+            statement.setLong(ID_INDEX, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete book with id: " + id, e);
+        }
+    }
+
+    private Book createBook(ResultSet resultSet) throws SQLException {
+        Book book = new Book();
+        book.setId(resultSet.getObject("id", Long.class));
+        book.setTitle(resultSet.getString("title"));
+        book.setPrice(resultSet.getObject("price", BigDecimal.class));
```

Comment:
```suggestion
        book.setPrice(resultSet.getBigDecimal("price"));
```

Reply:
In checklist, we have an example of best practice when we use getObject() method with Long variable. This practice share only on wrappers? And with others objects (BigDecimal, String) we can use its get method?